### PR TITLE
Fix layers import

### PR DESCRIPTION
**Pull request by p-l-**
This PR simplifies the way layers are loaded when running Scapy. It fixes Scapy under OpenBSD at least (currently broken). I also had to clean-up the imports in DHCPv6 layer.

### CODE REVIEW

**Comment by guedou** [comment]
Looks good. Could you add docstrigns to _load and `_load()` a,d `load_layer()`

**Comment by p-l-** [comment]
@guedou I did not modify `load_layers()`, so basically it works as it worked before (I think it would be great to add a docstring to this function, but also to a lot of other functions, maybe that could be done in other PRs).

As for `_load()`, I did adapt it so that we can use it in `scapy.layers.all`, but I am not sure to fully understand what it does and (more importantly) why. I can add a docstring, but if we want something more useful than `"_load(): loads stuff with black magic tricks"` I'm going to need help (and this, also, could be done in another PR).

File: scapy/main.py [hunk 1]
```diff
@@ -65,26 +68,33 @@ def _usage():
...
+                    globals_dict[name] = sym
+    except Exception:
+        log_interactive.error("Loading module %s", module, exc_info=True)
 
 def load_module(name):
     _load("scapy.modules."+name)
 
-def load_layer(name):
-    _load("scapy.layers." + LAYER_ALIASES.get(name, name))
+def load_layer(name, globals_dict=None, symb_list=None):
```
Comment (guedou) [inline scapy/main.py]: What do you think of ?
`"""Loads a Scapy layers to make variables, objects and functions available globally"""`

File: scapy/main.py [hunk 2]
```diff
@@ -65,26 +68,33 @@ def _usage():
 ######################
 
 
-def _load(module):
+def _load(module, globals_dict=None, symb_list=None):
```
Comment (guedou) [inline scapy/main.py]: `"""Loads a Python module to make variables, objects and functions available globally. The logic is to load the module using importlib, then copy the symbols to the global symbol table"""`

**Comment by guedou** [comment]
I made proposals for the docstring. I think that it is good hygiene to add docstrings when we touches some piece of code =)

What was broken on OpenBSD ?

**Comment by p-l-** [comment]
@guedou: I do agree it's good to have docstrings, I just didn't know what to write, so thanks a lot!

`./run_scapy` does not work because of import loops between `layers.{l2,inet}` and `arch.bpf`. This bug has been very hard to find (because of the use of `importlib` and the black magic with `globals()` and `__builtins__`), and quite hard to fix too...

File: scapy/arch/bpf/supersocket.py [hunk 3]
```diff
@@ -105,13 +107,16 @@ def set_promisc(self, value):
 
         try:
             fcntl.ioctl(self.ins, BIOCPROMISC, struct.pack('i', value))
-        except IOError as err:
-            msg = "Can't put your interface (%s) into promiscuous mode !" % self.iface
-            raise Scapy_Exception(msg)
+        except IOError:
+            raise Scapy_Exception("Cannott set promiscuous mode on interface "
```
Comment (guedou) [inline scapy/arch/bpf/supersocket.py]: s/Cannott/cannot/

Comment (p-l-) [reply scapy/arch/bpf/supersocket.py]: Fixed.

**Comment by guedou** [comment]
I works like a charm on OpenBSD 5.5 !

**Pull request merged**